SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -110,6 +111,28 @@ bool USpatialNetDriver::InitBase(bool bInitAsClient, FNetworkNotify* InNotify, c
 		return false;
 	}
 
+#if WITH_EDITOR
+	// If we're launching in PIE then ensure there is a deployment running before connecting.
+	FSpatialGDKServicesModule& GDKServices = FModuleManager::GetModuleChecked<FSpatialGDKServicesModule>("SpatialGDKServices");
+	FLocalDeploymentManager* LocalDeploymentManager = GDKServices.GetLocalDeploymentManager();
+
+	// Just connect if a deployment is running.
+	if (!LocalDeploymentManager->IsLocalDeploymentRunning() || LocalDeploymentManager->IsDeploymentStopping() || LocalDeploymentManager->IsDeploymentStarting())
+	{
+		UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Waiting for local spatial depoyment to start before connecting..."));
```

Comment:
typo

```suggestion
		UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Waiting for local spatial deployment to start before connecting..."));
```

Also surely if ` LocalDeploymentManager->IsDeploymentStopping()` is a case that lets execution enter this conditional, the `OnDeploymentStart` lambda will never be triggered because it's in the process of stopping not starting?

Reply:
Also surely if LocalDeploymentManager->IsDeploymentStopping() is a case that lets execution enter this conditional, the OnDeploymentStart lambda will never be triggered because it's in the process of stopping not starting?

Reply:
So when you press play you queue up a `TryStartLocalDeplyoment` basically. If the deployment is in the process of stopping then the `TryStartLocalDeplyoment` will wait for the current deployment to stop before starting a new one. Since the user is in the editor and has pressed the play button to enter this code path we can know if a deployment is stopping, another will start soon.